c

```diff
@@ -0,0 +1,259 @@
+	FILE* fpr = fopen("radnici.txt", "w");
+	if (fpr == NULL) {
+		printf("Greska pri otvaranju fajla\n");
+		return;
+	}
+	for (int i = 0; i < brRadnika; i++) {
+		fprintf(fpr, "%d %s %s %.2f %d\n", radnik[i].sifra, radnik[i].imeRadnika, radnik[i].prezimeRadnika, radnik[i].koeficijenat, radnik[i].radniStaz);
+	}
+	fclose(fpr);
+}
+
+void procitajRadnike(FILE* pfradnik, tipRadnika* radnik)
+{
+	fscanf(pfradnik, "%d", &radnik->sifra);
+	fscanf(pfradnik, "%s", radnik->imeRadnika);
+	fscanf(pfradnik, "%s", radnik->prezimeRadnika);
+	fscanf(pfradnik, "%.2f", &radnik->koeficijenat);
+	fscanf(pfradnik, "%d", &radnik->radniStaz);
+}
+
+void kopiraj_u_fajl(FILE* pftxt, tipRadnika radnik)
+{
+	fprintf(pftxt, "%d\n", radnik.sifra);
+	fprintf(pftxt, "%s\n", radnik.imeRadnika);
+	fprintf(pftxt, "%s\n", radnik.prezimeRadnika);
+	fprintf(pftxt, "%f\n", radnik.koeficijenat);
+	fprintf(pftxt, "%d\n", radnik.radniStaz);
+}
+
+void izmeniRadnika(tipRadnika* radnik, int odredjenaSifra) {
+
+	printf("Izmena podataka za radnika sa sifrom %d\n", radnik->sifra);
+	printf("Unesi novo ime radnika: \n");
+	scanf("%s", radnik->imeRadnika);
+	printf("Unesi novo prezime radnika: \n");
+	scanf("%s", radnik->prezimeRadnika);
+	printf("Unesi novi koeficijenat radnika: \n");
+	scanf("%f", &radnik->koeficijenat);
+	printf("Unesi novi radni staz radnika: \n");
+	scanf("%d", &radnik->radniStaz);
+
+}
+
+bool trazeniKoeficijenat(tipRadnika radnik, float trazeniKoef) {
+	return radnik.koeficijenat == trazeniKoef;
+}
+
+float izracunajTrenutnuZaradu(tipRadnika radnik, float trenutnaCenaRada) {
+	return radnik.koeficijenat * trenutnaCenaRada * (1 + radnik.radniStaz * 0.5 / 100);
+}
+
+int prebrojRadnikeSaTrazenimKoeficijentom(tipRadnika radnici[], int n, float trazeniKoef) {
+	int brojac = 0;
+	for (int i = 0; i < n; i++) {
+		if (radnici[i].koeficijenat == trazeniKoef)
+			brojac++;
+	}
+	return brojac;
+}
+
+int main()
+{
+
+	FILE* pr;
+	int brRadnika, izbor, redniBroj;
+	tipRadnika radnici[100];
+	char imeFajla[80];
+	int i, odaberiSifru, trazenaSifra;
+	float trazeniKoef, trenutnaCenaRada, trenutnaZarada;
+
+
+
+
+	while (1) {
+		puts("****************************************************");
+		puts("* MENI ");
+		puts("* 1 - Otvaranje tekstualnog fajla");
+		puts("* 2 - Snimanje radnika u tekstualni fajl");
+		puts("* 3 - Unos novog radnika");
+		puts("* 4 - Prikaz radnika(sa unetog rednog broja)");
+		puts("* 5 - Prikaz svih radnika");
+		puts("* 6 - Brisanje podataka o nekom radniku");
+		puts("* 7 - Izmena podataka o nekom radniku");
+		puts("* 8 - Provera da li radnik sa datim rednim brojem ima trazeni koeficijenat");
+		puts("* 9 - Izracunavanje zarade radnika sa datim rednim brojem");
+		puts("* 10 - Prebrojavanje radnika sa datim koeficijentom");
+		puts("* 0 - Kraj programa");
+
+
+		printf("Odaberi: \n");
+		scanf("%d", &izbor);
+
+		switch (izbor)
+		{
+
+		case 0:
+			//kraj fajla
+			pr = fopen("radnici.txt", "r");
+			if (pr != NULL)
+				return 0;
+			fclose(pr);
```

Comment:
Za opciju `0 - Kraj programa` ovde sto radis jeste otvaras fajl za citanje i zatvaras ga. Jesi li to zeleo da uradis ili je ovo bio samo placeholder kod ?

Reply:
Tu je poenta da kada korisnik ne zeli nista vise da radi unutar programa, da se sve sto unosi tokom koriscenja sacuva u fajl, ali mi je tu profesor pravio neko pitanje u vezi toga sto sam pisao i na kraju je ispalo ovo...

Reply:
Ukoliko zelis da u potpunosti zatvoris program, ono sto mozes da uradis je da sve to uklonis i ostavis samo

```c
return 0;
break;
```

To ce svakako da zavris izvrsenje programa.

Reply:
i cuva sve unutar fajla ?